### DISCUSSION TITLE
Meross MTS200 fails to connect to wifi using Custom Pairer App

### DISCUSSION THREAD

**Discussion by sr-pirotecnic** [krahabb/meross_lan#510] [General]
I'm using google nest wifi with 2.4+5GHz hybrid network , HA wired to it with and only one VLAN for my entire network. Had previously connected my thermosthat to network and HA using official Meross app but every few months the connection to HA would break and have to reconfigure again.

Tried connecting through Meross LAN broker using Custom Pairer app on Android (Pixel 7 Pro just in case). Steps followed:

- Reset the thermostat, holding the up and down arrows, several times
- Install Meross Local Broker, configure and start it in HA
- Login to broker using Custom Pairer app
- Try to connect with pairer app, it discovers the device but when entering wifi password it doesn't show as paired in the app
- Change wifi password temporarily as my wify password begins with *, then the device connects but does not show in Meross Local Broker Add-on in HA

The first problem with the password is a real inconvenience because I would have to reset all my wifi devices, including some wall hidden Sonoff Mini R2 flashed with ESPHome. Doable, but still an inconvenience.

The second problem I don't know how to fix, even changing the password wouldn't show the devices paired in the app to the Local Broker in HA.

**Comment by krahabb** [comment]
Hello @sr-pirotecnic,
Looking at the 'nomenclature' you're using, it looks as though you're using https://github.com/albertogeniola/ha-meross-local-broker which, beside some 'name affinity' is not `meross_lan` (this repo) and if this is the case you'd better off searching for support in that repository.
As far as my knowledge goes, there are actually 2 main projects (similar in some ways because they mostly achieve the same features) to unofficially support Meross devices in HA and both of these try their best to use local access with devices in order to make them less dependant on their 'by design' cloud nature.
Depending on youor needs you could be better off with either `ha-meross-local-broker` or `meross_lan`. I'm not too much into the features of the 'local broker' project but I guess it is a bigger step/project than `meross_lan` itself. This is because it goes thorugh re-pairing Meross devices to a different broker than the official ones and that step is often a bit painful. `meross_lan` on the other way could just 'side-communicate' locally (via HTTP) to Meross devices without the need to unpair them from the Meross cloud brokers. Of course this leaves devices exposed to the internet even though `meross_lan` only uses local traffic (the scenario might be more complicated though since meross_lan could also use, if configured so, the official brokers to reach to the devices and so can use both local and cloud protocols).
@albertogeniola (the owner of [ha-meross-local-broker](https://github.com/albertogeniola/ha-meross-local-broker) ) too built a component, similar to meross_lan, which is [meross-homeassistant](https://github.com/albertogeniola/meross-homeassistant) and is responsible to connect to Meross devices either locally or via cloud. As I've said, these are very similar in features (as always: to the best of my knowledge..)


So, in order to clarify a bit:
- If you just want to interface your Meross devices to HA (but leave them exposed to their App/cloud features) you can either use `meross_lan` or [meross-homeassistant](https://github.com/albertogeniola/meross-homeassistant). These two integration can both use local traffic (HTTP) and/or cloud traffic (cloud MQTT) when configured properly.
- If you want to completely disconnect the devices from the cloud brokers then `ha-meross-local-broker` is the way to go (or you can follow some hints [here](https://github.com/krahabb/meross_lan/discussions/63) on how to setup a local MQTT connection and pair the devices to this one so that `meross_lan` could use this instead of the official cloud brokers)
